core/src/dht/dht_reducers.rs

```diff
@@ -53,279 +50,90 @@ fn resolve_reducer(action_wrapper: &ActionWrapper) -> Option<DhtReducer> {
     }
 }
 
-pub(crate) fn reduce_hold_entry(
+pub(crate) fn reduce_commit_entry(
     context: Arc<Context>,
     old_store: &DhtStore,
     action_wrapper: &ActionWrapper,
 ) -> Option<DhtStore> {
-    match action_wrapper.action().clone() {
-        Action::Commit((entry, _, _)) => reduce_store_entry_common(context, old_store, &entry),
-        Action::Hold(EntryWithHeader { entry, header }) => {
-            reduce_store_entry_common(context.clone(), old_store, &entry).and_then(|state| {
-                state.add_header_for_entry(&entry, &header).ok()?;
-                Some(state)
-            })
+    let (entry, _, _) = unwrap_to!(action_wrapper.action() => Action::Commit);
+    let mut new_store = (*old_store).clone();
+    match reduce_store_entry_inner(&mut new_store, entry) {
+        Ok(()) => Some(new_store),
```

Comment:
Hm, I don't think it makes sense to add an entry to the DHT store without its header. Either we don't store entries here on commit or we store them the same way as we would when holding. So I would suggest we stick to reducing `Action::Commit` with `reduce_hold_entry`.

What was the reasoning behind separating the handling of commit actions like this?

Reply:
This was already the case I just refactored it to make it more explicit. While there was a single reducer for both `Action::Commit` and `Action::Hold` the reducer matched on the action type and only added the header in the hold case. [See here](https://github.com/holochain/holochain-rust/blob/c065bbdbc4af29ac7f837efb1531a6011b60f86d/core/src/dht/dht_reducers.rs#L61).

If we do want to change how it works it might be worth making a separate PR to discuss it

Reply:
Oh, wow. Ok...